source/opt/generate_webgpu_initializers_pass.cpp

```diff
@@ -92,27 +88,24 @@ Pass::Status GenerateWebGPUInitializersPass::Process() {
 Instruction* GenerateWebGPUInitializersPass::GetNullConstantForVariable(
     Instruction* variable_inst) {
   auto constant_mgr = context()->get_constant_mgr();
-  auto* constant_type = context()
-                            ->get_constant_mgr()
-                            ->GetType(variable_inst)
-                            ->AsPointer()
-                            ->pointee_type();
-
-  if (null_constant_type_map_.find(constant_type) ==
-      null_constant_type_map_.end()) {
+  auto* def_use_mgr = get_def_use_mgr();
+
+  auto* ptr_inst = def_use_mgr->GetDef(variable_inst->type_id());
+  auto type_id = ptr_inst->GetInOperand(1).words[0];
+  if (null_constant_type_map_.find(type_id) == null_constant_type_map_.end()) {
+    auto* constant_type = context()->get_type_mgr()->GetType(type_id);
     auto* constant = constant_mgr->GetConstant(constant_type, {});
```

Comment:
I missed this before, but you should provide the type_id here (its an optional parameter).